src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/CreateCollectionOperation.java

```diff
@@ -235,12 +236,21 @@ private Uni<Supplier<CommandResult>> executeCollectionCreation(
               }
               // if index creation violates DB index limit and collection existed before,
               // will not drop the collection
-              return Uni.createFrom()
-                  .item(
-                      () ->
-                          ErrorCode.TOO_MANY_INDEXES.toApiException(
-                              "collection \"%s\" creation failed due to index creation failing; need %d indexes to create the collection;",
-                              name, dbLimitsConfig.indexesNeededPerCollection()));
+              if (error.getMessage().matches("Index .* already exists")) {
+                return Uni.createFrom()
+                    .item(
+                        () ->
+                            ErrorCode.INDEXES_CREATION_FAILED.toApiException(
```

Comment:
-1 on this error message, it hides information the API already knows. 

The API knows the reason the index failed to be created was that it already existed, we should say that in the message. In general avoid messages that only say "it did not work" 

Also, the "check schema" part - what should the user do ? Check it for what ? We should give actionable advice.  

Same on line 250 below.